digdag-core/src/main/java/io/digdag/core/schedule/ScheduleExecutor.java

```diff
@@ -335,41 +377,97 @@ private Scheduler getSchedulerOfSchedule(StoredSchedule sched)
             }
 
             // run sessions
-            ImmutableList.Builder<StoredSessionAttemptWithSession> attempts = ImmutableList.builder();
-            for (Instant instant : instants) {
-                if (dryRun) {
-                    attempts.add(
-                            StoredSessionAttemptWithSession.dryRunDummy(siteId,
-                                Session.of(def.getProject().getId(), def.getName(), instant),
-                                ImmutableStoredSessionAttempt.builder()
-                                    .retryAttemptName(Optional.of(attemptName))
-                                    .workflowDefinitionId(Optional.of(def.getId()))
-                                    .timeZone(def.getTimeZone())
-                                    .id(0L)
-                                    .params(def.getConfig().getFactory().create())
-                                    .stateFlags(AttemptStateFlags.empty())
-                                    .sessionId(0L)
-                                    .createdAt(Instant.now())
-                                    .finishedAt(Optional.absent())
-                                    .build()
-                            )
-                        );
-                }
-                else {
-                    try {
-                        StoredSessionAttemptWithSession attempt = handler.start(def,
-                                ScheduleTime.of(instant, sched.getNextScheduleTime()),
-                                Optional.of(attemptName), sched.getLastSessionTime());
-                        attempts.add(attempt);
-                        // TODO this may throw ResourceLimitExceededException. But some sessions are already committed. To be able to rollback everything, inserting all sessions needs to be in a single transaction.
+            return workflowExecutor.submitTransaction(siteId, (submitter) -> {
```

Comment:
The above line https://github.com/treasure-data/digdag/pull/431/files#diff-a0e8a0113728f65511442e76d9e406adR339 can be removed

```
            ScheduleControl lockedSched = new ScheduleControl(store, sched);
```